jupyter_server/services/api/handlers.py

```diff
@@ -110,9 +111,30 @@ async def get(self):
         model = {
             "identity": identity,
             "permissions": permissions,
+            "updatable_fields": self.identity_provider.updatable_fields,
         }
         self.write(json.dumps(model))
 
+    @web.authenticated
+    async def patch(self):
+        """Update user information."""
+        user_data = cast(dict[UpdatableField, str], self.get_json_body())
+        if not user_data:
+            raise web.HTTPError(400, "Invalid or missing JSON body")
+
+        # Update user information
+        identity_provider = self.settings["identity_provider"]
+        if not isinstance(identity_provider, IdentityProvider):
+            raise web.HTTPError(500, "Identity provider not configured properly")
+
+        try:
+            updated_user = identity_provider.update_user(self, user_data)
```

Comment:
I would probably pass `self.current_user` here rather than `self`, so it's not part of the `update_user` API how we store the current user on the Handler. Then pass `self on the end for IdentityProviders that need to use Handler for persistence (some won't, e.g. JupyterHub).